src/argilla/client/feedback.py

```diff
@@ -0,0 +1,809 @@
+        FeedbackDatasetModel,
+        FeedbackFieldModel,
+        FeedbackItemModel,
+        FeedbackQuestionModel,
+    )
+
+FETCHING_BATCH_SIZE = 250
+
+
+class ValueSchema(BaseModel):
+    value: Union[StrictStr, StrictInt]
+
+
+class ResponseSchema(BaseModel):
+    values: Dict[str, ValueSchema]
+    status: Literal["submitted", "missing", "discarded"]
+
+
+class OfflineFeedbackResponse(ResponseSchema):
+    user_id: Optional[UUID] = None
+
+    @validator("user_id", always=True)
+    def user_id_must_have_value(cls, v):
+        if not v:
+            warnings.warn(
+                "`user_id` not provided for `OfflineFeedbackResponse`. So it will be"
+                " set to `None`. Which is not an issue, even though if you're planning"
+                " to log the record in Argilla, as it will be set automatically to the"
+                " current user's id."
+            )
+        return v
+
+
+class FeedbackRecord(BaseModel):
+    fields: Dict[str, str]
+    responses: Optional[Union[ResponseSchema, List[ResponseSchema]]] = None
+    external_id: Optional[str] = None
+
+    @validator("responses", always=True)
+    def responses_must_be_a_list(cls, v):
+        if not v:
+            return []
+        if isinstance(v, ResponseSchema):
+            return [v]
+        return v
+
+    class Config:
+        extra = Extra.forbid
+
+
+class OfflineFeedbackRecord(BaseModel):
+    id: Optional[str] = None
+    fields: Dict[str, str]
+    responses: List[OfflineFeedbackResponse] = []
+    external_id: Optional[str] = None
+    inserted_at: Optional[str] = None
+    updated_at: Optional[str] = None
+
+
+class FieldSchema(BaseModel):
+    name: str
+    title: Optional[str] = None
+    required: Optional[bool] = True
+    settings: Dict[str, Any]
+
+    @validator("title", always=True)
+    def title_must_have_value(cls, v, values):
+        if not v:
+            return values["name"].capitalize()
+        return v
+
+
+class TextField(FieldSchema):
+    settings: Dict[str, Any] = Field({"type": "text"}, const=True)
+
+
+FIELD_TYPE_TO_PYTHON_TYPE = {"text": str}
+
+
+class QuestionSchema(BaseModel):
+    name: str
+    title: Optional[str] = None
+    description: Optional[str] = None
+    required: Optional[bool] = True
+    settings: Dict[str, Any]
+
+    @validator("title", always=True)
+    def title_must_have_value(cls, v, values):
+        if not v:
+            return values["name"].capitalize()
+        return v
+
+
+# TODO(alvarobartt): add `TextResponse` and `RatingResponse` classes
+class TextQuestion(QuestionSchema):
+    settings: Dict[str, Any] = Field({"type": "text"}, const=True)
+
+
+class RatingQuestion(QuestionSchema):
+    settings: Dict[str, Any] = Field({"type": "rating"})
+    values: List[int]
+
+    @validator("values", always=True)
+    def update_settings_with_values(cls, v, values):
+        if v:
+            values["settings"]["options"] = [{"value": value} for value in v]
+        return v
+
```

Comment:
If we want to keep updated values and settings once the question object is created, we should enable the `validate_assignment` config attribute. See [docs](https://docs.pydantic.dev/latest/usage/model_config/#options) 
```suggestion
class RatingQuestion(QuestionSchema):
    settings: Dict[str, Any] = Field({"type": "rating"})
    values: List[int]

    @validator("values", always=True)
    def update_settings_with_values(cls, v, values):
        if v:
            values["settings"]["options"] = [{"value": value} for value in v]
        return v
    
    class Config:
        validate_assignment= True
```

Reply:
You mean to do something like e.g.

```python
import argilla as rg

question = rg.RatingQuestion(name="rating", required=True, values=[1, 2, 3])
question.values = [1, 2, 3, 4, 5]
```

Not sure if that's what you mean

Reply:
Exactly

Reply:
Ok then I'll update that!